src/compiler/chainrules.jl

```diff
@@ -18,19 +17,45 @@ such that if a suitable rule is defined later, the generated function will recom
 """
 function has_chain_rrule(T)
   config_T, arg_Ts = Iterators.peel(T.parameters)
-  m_with_config = meta(Tuple{typeof(rrule), config_T, arg_Ts...})
-  if m_with_config.method === rrule_redispatcher_method
-    # it is being redispatched without config, so check it that hits the fallback
-    m_without_config = meta(Tuple{typeof(rrule), arg_Ts...})
-    if m_without_config.method === rrule_fallback_method
-      # no rrule exists, return instance for m_with_config as that will be invalidated 
-      # directly if configured rule added, or indirectly if unconfigured rule added
-      return false, m_with_config.instance
-    end
+  configured_rrule_m = meta(Tuple{typeof(rrule), config_T, arg_Ts...})
+  if _is_rrule_redispatcher(configured_rrule_m.method)
+    # it is being redispatched without config, so get the method it redispatches to
+    rrule_m = meta(Tuple{typeof(rrule), arg_Ts...})
+    no_rrule_m = meta(Tuple{typeof(ChainRulesCore.no_rrule), arg_Ts...})
+  else
+    # Not being redispatched
+    rrule_m = configured_rrule_m
+    no_rrule_m = meta(Tuple{typeof(ChainRulesCore.no_rrule), config_T, arg_Ts...})
+  end
+
+  do_not_use_rrule = matching_cr_sig(no_rrule_m, rrule_m)
+  if do_not_use_rrule
```

Comment:
Is this meant to do decomposition?

Reply:
This function is `has_chain_rrule` and if it returns `false,...` then `pullback` will indeed end up calling `generate_pullback_via_decomposition`.